### PR TITLE
Switch pool creation to use size_t, not int in C API

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: src/umpire/interface/c_fortran/typesUmpire.h [hunk 1]
```diff
@@ -1,9 +1,3 @@
-//////////////////////////////////////////////////////////////////////////////
-// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC and Umpire
-// project contributors. See the COPYRIGHT file for details.
-//
-// SPDX-License-Identifier: (MIT)
-//////////////////////////////////////////////////////////////////////////////
```
Comment (mcfadden8) [inline src/umpire/interface/c_fortran/typesUmpire.h]: Did some tools change that caused this copyright to go missing?

File: src/umpire/interface/c_fortran/typesUmpire.h [hunk 2]
```diff
@@ -30,11 +24,11 @@
 extern "C" {
 #endif
 
-struct s_UMP_SHROUD_capsule_data {
+struct s_umpire_SHROUD_capsule_data {
```
Comment (mcfadden8) [inline src/umpire/interface/c_fortran/typesUmpire.h]: Will these name changes (from `s_UMP_` to `s_umpire_`) cause any externally visible API changes?

File: src/umpire/interface/c_fortran/wrapResourceManager.cpp [hunk 3]
```diff
@@ -214,7 +208,7 @@ size_t umpire_resourcemanager_get_size(umpire_resourcemanager * self,
 
 umpire_allocator * umpire_resourcemanager_make_allocator_pool(
     umpire_resourcemanager * self, const char * name,
-    umpire_allocator allocator, int initial_size, int block,
+    umpire_allocator allocator, size_t initial_size, size_t block,
```
Comment (mcfadden8) [inline src/umpire/interface/c_fortran/wrapResourceManager.cpp]: Shouldn't these be `std::size_t`?

File: src/umpire/interface/c_fortran/typesUmpire.h (hunk 2 above)
Comment (davidbeckingsale) [reply src/umpire/interface/c_fortran/typesUmpire.h]: Nope, these are internal.

File: src/umpire/interface/c_fortran/typesUmpire.h (hunk 1 above)
Comment (davidbeckingsale) [reply src/umpire/interface/c_fortran/typesUmpire.h]: Yup, I can fix that.

File: src/umpire/interface/c_fortran/wrapResourceManager.cpp (hunk 3 above)
Comment (davidbeckingsale) [reply src/umpire/interface/c_fortran/wrapResourceManager.cpp]: Nope, because this is a C function, so no namespacing.

Comment (mcfadden8) [reply src/umpire/interface/c_fortran/wrapResourceManager.cpp]: Then shouldn't this file be renamed with a `.c` suffix instead of of `.cpp`?

**Comment by davidbeckingsale** [comment]
@mcfadden8 - this is ready to go.

File: src/umpire/interface/c_fortran/typesUmpire.h [hunk 4]
```diff
@@ -30,11 +30,11 @@
...
```
Comment (mcfadden8) [inline src/umpire/interface/c_fortran/typesUmpire.h]: Should the header of this file change too?

Comment (davidbeckingsale) [reply src/umpire/interface/c_fortran/typesUmpire.h]: I'm not sure I understand what you are saying.

**Comment by davidbeckingsale** [comment]
I don't think tests allocating >4GiB will pass on Travis/Azure.

**Comment by mcfadden8** [comment]
We have tests running for devices on bamboo.  If we added some device testing, that should cover it.

File: src/umpire/interface/c_fortran/typesUmpire.h (hunk 4 above)
Comment (mcfadden8) [reply src/umpire/interface/c_fortran/typesUmpire.h]: The copyright/boilerplate header for this file seems different.

**Pull request merged**